chainerx_cc/chainerx/cuda/memory_pool_test.cc

```diff
@@ -356,6 +357,26 @@ TEST(MemoryPoolTest, FreeUnusedBlocksSplitAndFreeHead) {
     EXPECT_EQ(allocator->free_called(), 0);
 }
 
+TEST(MemoryPoolTest, FreeUnusedBlocksThreadSafe) {
```

Comment:
A minor but how about using the TEST_THREAD_SAFE macro since it's meant for these kinds of tests? Although the code would be very similar right now, we could add features to the macro later on such as repeat counts.

Reply:
Discussed offline to add a TODO comment instead since the thread sanitizer isn't really working with CUDA.

Reply:
Added TODO.